sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/TableCatalogSuite.scala

```diff
@@ -379,27 +379,12 @@ class TableCatalogSuite extends SparkFunSuite {
     assert(table.schema == originalSchema)
 
     val updated = catalog.alterTable(testIdent,
-      TableChange.updateColumnType(Array("id"), LongType, true))
+      TableChange.updateColumnNullability(Array("id"), true))
 
     val expectedSchema = new StructType().add("id", LongType).add("data", StringType)
     assert(updated.schema == expectedSchema)
   }
 
-  test("alterTable: update optional column to required fails") {
```

Comment:
we do the check at Spark side now, catalog implementation doesn't need to care about it. Besides, the catalog implementation being tested doesn't check the safety of the type change either, because it's done at spark side.